test/Dialect/Calyx/emit-custom-primitives.mlir

```diff
@@ -0,0 +1,40 @@
+// RUN: circt-translate --export-calyx --verify-diagnostics %s | FileCheck %s --strict-whitespace
```

Comment:
Similarly, it would be nice to verify some of the errors you've spelled out in https://github.com/llvm/circt/blob/main/test/Dialect/Calyx/errors.mlir